friends/src/components/FriendCard.js

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+// import { Link } from "react-router-dom";
+import axios from "axios";
+import UpdateFriend from "./UpdateFriend";
+import {
+  Card,
+  CardText,
+  CardBody,
+  CardTitle,
+  CardSubtitle,
+} from "reactstrap";
+
+export default class FriendCard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      userInfo: null
+    };
+  }
+
+  componentDidMount() {
+    const id = this.props.match.params.id;
+    this.fetchFriend(id);
+  }
+
+  fetchFriend = id => {
+    axios
+      .get(`http://localhost:5000/friends`)
+      .then(response => {
+        const found = response.data.find(friend => `${friend.id}` === id);
+        this.setState(() => ({ userInfo: found }));
+      })
+      .catch(error => {
+        console.error(error);
+      });
+  };
+
+  inputChange = e => {
+    e.preventDefault();
+    this.setState({
+      userInfo: {
+        ...this.state.userInfo,
+        [e.target.name]: e.target.value
+      }
+    });
+  };
+
+  numberInputChange = e => {
```

Comment:
If you set the `type` in your form field to `number` you may not need this handler.